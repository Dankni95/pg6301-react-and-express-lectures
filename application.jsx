import React from "react";

export function Application({who}) {
  return <h1>Hello {who}</h1>;
}
