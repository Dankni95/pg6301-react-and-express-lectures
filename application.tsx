import * as React from "react";
import { BrowserRouter, Link, Route, Routes } from "react-router-dom";

function FrontPage() {
  return (
    <div>
      <h1>Movie database</h1>

      <ul>
        <li>
          <Link to={"/movies"}>List movies</Link>
        </li>
        <li>
          <Link to={"/movies/new"}>Add movie</Link>
        </li>
      </ul>
    </div>
  );
}

export function Movies() {
  return (
    <Routes>
      <Route path="/" element={<h1>All movies</h1>} />
      <Route path="/new" element={<h1>Add movie</h1>} />
    </Routes>
  );
}

export function Application() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path={"/"} element={<FrontPage />} />
        <Route path={"/movies/*"} element={<Movies />} />
        <Route path={"/*"} element={<h1>Not found</h1>} />
      </Routes>
    </BrowserRouter>
  );
}
