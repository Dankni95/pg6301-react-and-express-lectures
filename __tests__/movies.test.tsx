import React from "react";
import ReactDOM from "react-dom";
import { Movies } from "../application";
import { MemoryRouter } from "react-router-dom";

describe("movies", () => {
  it("renders movie list", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Movies />
      </MemoryRouter>,
      element
    );
    expect(element).toMatchSnapshot();
  });
  it("renders new movie form", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter initialEntries={["/new"]}>
        <Movies />
      </MemoryRouter>,
      element
    );
    expect(element).toMatchSnapshot();
  });
});
