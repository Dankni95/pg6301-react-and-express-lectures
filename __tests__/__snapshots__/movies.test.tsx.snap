// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies renders movie list 1`] = `
<div>
  <h1>
    All movies
  </h1>
</div>
`;

exports[`movies renders new movie form 1`] = `
<div>
  <h1>
    Add movie
  </h1>
</div>
`;
